### PR TITLE
Extra args sharing with link

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Added kwargs, expiration date is not ready in OCS Sharing API yet.

### CODE REVIEW

File: owncloud/owncloud.py [hunk 1]
```diff
@@ -652,25 +652,41 @@ def copy(self, remote_path_source, remote_path_target):
...
         :param path: path to the remote file to share
+        :param perms (optional): permission of the shared object
+        defaults to read only (1)
+        :param public_upload (optional): allows users to upload files or folders
+        :param password (optional): sets a password
+        http://doc.owncloud.org/server/6.0/admin_manual/sharing_api/index.html
         :returns: instance of :class:`PublicShare` with the share info
             or False if the operation failed
-        :raises: HTTPResponseError in case an HTTP error status was returned
+        :raises: ResponseError in case an HTTP error status was returned
```
Comment (PVince81) [inline owncloud/owncloud.py]: Please use the new HTTPResponseError. Maybe you forgot to pull master ?

**Comment by PVince81** [comment]
Thanks. Do you mind adding or updating the matching unit tests ?

File: owncloud/owncloud.py (hunk 1 above)
Comment (SergioBertolinSG) [inline owncloud/owncloud.py]: Changed already not sure why it is not appearing here as outdated diff.

**Comment by SergioBertolinSG** [comment]
I've modified the test regarding sharing with link, but I'm not sure if assertions are needed to check public availability or password also I don't know how to check that.

File: owncloud/owncloud.py (hunk 1 above)
Comment (PVince81) [inline owncloud/owncloud.py]: Did you forget to commit/push maybe ?
You can compare your local `git log` with the commit list from this PR

**Comment by PVince81** [comment]
Never mind the password change then.
You could add some tests that update the "public upload" and "perms" and then try and re-read them from the server to see if they match.

**Comment by PVince81** [comment]
Sorry, wrong click

**Comment by SergioBertolinSG** [comment]
public shares currently only returns share_id, path, link and token. So it cannot be checked until improvements in core.

Example:
PublicShare(id=142,path=/Photos,link=http://OC_URL/index.php/s/sCAtEP3peiMn4LS,token=sCAtEP3peiMn4LS)

**Comment by PVince81** [comment]
Okay, so this is untestable.

Please double check the HttpResponseError part.

**Comment by SergioBertolinSG** [comment]
If you see the full file, you can see the HTTPResponse there.

**Comment by PVince81** [comment]
I see ResponseError here https://github.com/SergioBertolinSG/pyocclient/blob/extra_args_sharing_with_link/owncloud/owncloud.py#L666

**Comment by SergioBertolinSG** [comment]
Anything else needed?

**Comment by PVince81** [comment]
Looks good :+1:

**Pull request merged**